subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorNestingIntegrationTest.groovy

```diff
@@ -146,6 +146,15 @@ class WorkerExecutorNestingIntegrationTest extends AbstractWorkerExecutorIntegra
 
         buildFile << """
             ${getWorkActionWithNesting("noIsolation", "noIsolation")}
+            class WorkerExecutorThreads {
+                static def getWorkerExecutorThreads() {
+                    def threadGroup = Thread.currentThread().threadGroup
+                    def threads = new Thread[threadGroup.activeCount()]
+                    threadGroup.enumerate(threads)
+                    return threads.findAll { it?.name?.startsWith("${WorkerExecutionQueueFactory.QUEUE_DISPLAY_NAME}") }
+                }
+            }
```

Comment:
❓ Can we define the original method as `static` instead of creating this wrapper class?

Reply:
That fails, according to the recommendations in the [docs](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:not_yet_implemented:accessing_top_level_at_execution) a top-level method must be converted to a static method within a class.